libraries/AP_Avoidance/AP_Avoidance.cpp

```diff
@@ -384,10 +384,7 @@ void AP_Avoidance::update_threat_level(const Location &my_loc,
 }
 
 MAV_COLLISION_THREAT_LEVEL AP_Avoidance::current_threat_level() const {
-    if (_obstacles == nullptr) {
-        return MAV_COLLISION_THREAT_LEVEL_NONE;
-    }
-    if (_current_most_serious_threat == -1) {
+    if (_obstacles == nullptr || _current_most_serious_threat == -1) {
```

Comment:
Please don't coalesce these.  I have mentioned this before.  It all starts off great - until nobody can read the if-clause.

Reply:
@peterbarker san..
I'm coding with sequence diagrams and flowchart diagrams. Writing the same process multiple times is not good code. There may be many conditions.

Reply:
@peterbarker san..
I'm coding with sequence diagrams and flowchart diagrams. Writing the same process multiple times is not good code. There may be many conditions.

Reply:
I'm in favor of grouping them